src/Gaffer/Plug.cpp

```diff
@@ -837,11 +840,11 @@ class Plug::DirtyPlugs
 			}
 
 			VertexDescriptor result = add_vertex( m_graph );
-			m_graph[result] = const_cast<Plug *>( plug );
+			m_graph[result] = plug;
 			m_plugs[plug] = result;
 
 			// Insert parent plug.
-			if( const Plug *parent = plug->parent<Plug>() )
+			if( auto parent = plug->parent<Plug>() )
```

Comment:
Curious about where you're at with `auto`?  This seems like a case where I would find it clearer to just say `Plug *`?  I'm just curious how I should be approaching this.

Reply:
Some days I feel more `auto` than others, but I think there are two places I'm using it fairly consistently :

- Where my head would explode with template/typename nonsense otherwise.
- Where the type is already present on the line, as in `parent<Plug>()` above.

Reply:
You might find this interesting : an argument auto should be used a _lot_ :

https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/

I don't remember the details, but I didn't find it completely convincing for some reason, probably my age :)